public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx

```diff
@@ -281,6 +283,49 @@ const isCortexLokiOrRecordingRule = (watch: UseFormWatch<RuleFormValues>) => {
   return (ruleType === RuleFormType.cloudAlerting || ruleType === RuleFormType.cloudRecording) && dataSourceName !== '';
 };
 
+// the backend will always execute "hidden" queries, so we have no choice but to remove the property in the front-end
+// to avoid confusion. The query editor shows them as "disabled" and that's a different semantic meaning.
+// furthermore the "AlertingQueryRunner" calls `filterQuery` on each data source and those will skip running queries that are "hidden"."
+// It seems like we have no choice but to act like "hidden" queries don't exist in alerting.
+const ignoreHiddenQueries = (ruleDefinition: RuleFormValues): RuleFormValues => {
+  return {
+    ...ruleDefinition,
+    queries: ruleDefinition.queries?.map((query) => omit(query, 'model.hide')),
+  };
+};
+
+function formValuesFromQueryParams(ruleDefinition: string, type: RuleFormType): RuleFormValues {
+  let ruleFromQueryParams: Partial<RuleFormValues>;
+
+  try {
+    ruleFromQueryParams = JSON.parse(ruleDefinition);
+  } catch (err) {
+    return {
+      ...getDefaultFormValues(),
+      queries: getDefaultQueries(),
+    };
+  }
+
+  return ignoreHiddenQueries({
+    ...getDefaultFormValues(),
+    ...ruleFromQueryParams,
+    queries: ruleFromQueryParams.queries ?? getDefaultQueries(),
```

Comment:
regarding the comment in line 290, in here we could call the ignoreHiddenQueries and only transform the queries array instead of the whole RuleFormValues object